### PR TITLE
#707 Removed use of any in projects.api

### LINKED ISSUE #707: [Refactoring] - Remove `any` type from projects.api
### Description

Remove the use of the `any` type from projects.api. This may seem very simple, but the point of this is just to have you read through the codebase and understand the types/interfaces used in it, or make new ones that fit the shape of the data being used.

### Notes

Check #665 for notes on why we're doing this and pointers on what you should replace `any` with.

### Acceptance Criteria

`any` is not used as a type in projects.api.

### Proposed Solution

`Ctrl + F`

### PR DESCRIPTION
**Pull request by Lomloam**
## Changes

Replaced any with CreateSingleObjectPayload and EditSingleObjectPayload, both interfaces

## Notes

Also replaced two instances of any in project.hooks where editSingleObject and createSingleObject were involved

## Checklist

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #707 #687

### CODE REVIEW

File: src/frontend/src/apis/projects.api.ts [hunk 1]
```diff
@@ -45,7 +46,7 @@ export const createSingleProject = (payload: any) => {
  *
  * @param payload Payload containing all information needed to edit a project.
  */
-export const editSingleProject = (payload: any) => {
+export const editSingleProject = (payload: EditSingleObjectPayload) => {
```
Comment (Peyton-McKee) [inline src/frontend/src/apis/projects.api.ts:49]: editSingleProject

File: src/frontend/src/hooks/projects.hooks.ts [hunk 2]
```diff
@@ -40,10 +41,13 @@ export const useSingleProject = (wbsNum: WbsNumber) => {
  *
  */
 export const useCreateSingleProject = () => {
-  return useMutation<{ message: string }, Error, any>(['projects', 'create'], async (projectPayload: any) => {
-    const { data } = await createSingleProject(projectPayload);
-    return data;
-  });
+  return useMutation<{ message: string }, Error, any>(
```
Comment (Peyton-McKee) [inline src/frontend/src/hooks/projects.hooks.ts:44]: make this any of type CreateSingleProjectPayload

File: src/frontend/src/hooks/projects.hooks.ts [hunk 3]
```diff
@@ -53,7 +57,7 @@ export const useEditSingleProject = (wbsNum: WbsNumber) => {
   const queryClient = useQueryClient();
```
Comment (Peyton-McKee) [inline src/frontend/src/hooks/projects.hooks.ts:57]: make this editSingleProjectPayload

File: src/frontend/src/utils/types.ts [hunk 4]
```diff
@@ -25,3 +25,28 @@ export interface LinkItem {
 export interface VersionObject {
   tag_name: string;
 }
+
+export interface EditSingleObjectPayload {
```
Comment (Peyton-McKee) [inline src/frontend/src/utils/types.ts:29]: projectPayload

**Review by Peyton-McKee — CHANGES_REQUESTED** [review]
Great Work! Just some naming conventions and make sure to remove the anys on the hooks!

**Review by Peyton-McKee — APPROVED** [review]
lgtm

**Review by anthonybernardi — APPROVED** [review]
awesome

**Pull request merged**